libs/s25main/world/GameWorldBase.h

```diff
@@ -79,7 +79,7 @@ class GameWorldBase : public World
     // Remaining initialization after loading (BQ...)
     void InitAfterLoad();
 
-    /// Setzt GameInterface
+    GameInterface* GetGameInterface() const { return gi; }
```

Comment:
In principle I would agree with this, but in practice there will be other classes that will want to `GetCheats()` but only have access to a `const GameWorldBase` (such as [GamePlayer](https://github.com/kubaau/s25client/commit/661d1f41b042ec830ba17017a6037d64fd63acb6?diff=unified#diff-52f6aea2a82a922f095f592a2f1b53effc939822718441624200437cbbf7544dR2260) and [GameWorldView](https://github.com/kubaau/s25client/commit/395596ff2985d239b9521f631d325bb18af01520?diff=unified#diff-ed4c289e07ca0a90c1cd9f11b1fc2e93327a11ea1c01fe53800d7a6540b7d071R377)). Creating two versions like this:
```
    GameInterface* GetGameInterface() { return gi; }
    const GameInterface* GetGameInterface() const { return gi; }
```
will also force to extend GameInterface like this:
```
    virtual const Cheats& GI_GetCheats() const = 0;
    virtual Cheats& GI_GetCheats() = 0;
```
or like this:
```
    virtual Cheats& GI_GetCheats() const = 0;
```
Where the first solution only provides a minor safety improvement and the second just moves the issue someplace else. It is simply convenient to have it defined like this.

Reply:
> Creating two versions like this: [..] will also force to extend GameInterface like this:

Yes that is what I had in mind: `Cheats` is used to query some state (read-only) and modify state. Those with access to `const GameWorldBase` should not modify the state and both places you mentioned don't do that. So from a `const GameWorldBase` you only get a `const GameInterface` and from that only a `const Cheats`. That's how I understand "const correctness".

To avoid 2 virtuals, you can "cheat": 
```
const Cheats& GI_GetCheats() const = 0;
virtual Cheats& GI_GetCheats()  { const auto* me = this; return const_cast<Cheats&>(me->GI_GetCheats()); }
```

Not sure what you mean by "only minor safety improvement". E.g. `std::vector` and most other data structures work like this. And yes const correctness provides only a small advantage. But when it catches an unintentional access/modification it is quite useful, isn't it? And I'd rather do it consistently "right" for newly added code, see https://isocpp.org/wiki/faq/const-correctness#retrofitting-const

Reply:
Added what was requested, just implemented the virtual hack differently.